src/main/java/org/broadinstitute/hellbender/tools/reference/ReferenceSequenceTable.java

```diff
@@ -0,0 +1,437 @@
+                }
+            }
+        }
+        return refPairs;
+    }
+
+    @Override
+    public Iterator<TableRow> iterator() {
+        checkTableBuildStatus();
+
+        return tableByMD5.values().iterator();
+    }
+
+    /**
+     * Minimal class representing a table entry within a row of the ReferenceSequenceTable.
+     * Stores the entry's value and the associated column.
+     */
+    public class TableEntry {
+
+        private final String columnName;
+        private final String columnValue;
+
+        public TableEntry(String columnName, String columnValue){
+            this.columnName = columnName;
+            this.columnValue = columnValue;
+        }
+
+        public String getColumnName() { return columnName; }
+        public String getColumnValue() { return columnValue; }
+
+        public boolean isEmpty(){
+            return columnValue.equals(MISSING_ENTRY_DISPLAY_STRING);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            TableEntry that = (TableEntry) o;
+            return columnName.equals(that.columnName) && columnValue.equals(that.columnValue);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(columnName, columnValue);
+        }
+    }
+
+    /**
+     * Class representing a row of the ReferenceSequenceTable. Rows are keyed by MD5 and the remaining columns
+     * contain the sequence length, followed by the sequence name across each reference.
+     */
+    public class TableRow {
+        private final String md5;
+        private final TableEntry[] entries;
+        private final int length;
+
+        public TableRow(String md5, int length, List<GATKPath> references) {
+            this.md5 = md5;
+            this.length = length;
+            entries = new TableEntry[references.size()+2];
+
+            add(new TableEntry(MD5_COLUMN_NAME, md5));
+            add(new TableEntry(LENGTH_COLUMN_NAME, Integer.toString(length)));
+
+            for(int i = LENGTH_COLUMN_INDEX + 1; i < entries.length; i++){
+                entries[i] = new TableEntry(getColumnNames().get(i), MISSING_ENTRY_DISPLAY_STRING);
+            }
+        }
+
+        public void add(TableEntry entry) {
+            int idx = columnIndices.get(entry.getColumnName());
+            entries[idx] = entry;
+        }
+
+        public String getMd5() {
+            return md5;
+        }
+
+        public TableEntry[] getEntries() {
+            return entries;
+        }
+
+        public int getLength() {
+            return length;
+        }
+
+        public List<String> getColumnNames() {
+            return columnNames;
+        }
+
+        public String toString(){
+            String row = String.format("%s\t", this.md5);
+            for(int i = 0; i < entries.length; i++){
+                row += entries[i].columnValue;
```

Comment:
Does the `toString()` method end up adding the MD5 twice, since it's in column 0?